Reduce log level for Progress API

hubertp opened the pull request:
### Pull Request Description

When showing trace log levels, logs are polluted with progress info. Unnecessarily.
Progress updates are still being shown.
Also fixed a bug when configuration was provided via env variable and an entry in the config file (the latter should be ignored).
Closes #13368.

hubertp committed:
Reduce log level for Progress API

When showing trace log levels, logs are polluted with progress info.
Unnecessarily.

hubertp committed:
Update log config for runner

Same as for Language Server

hubertp committed:
Do not send credentials to appenders

Not particularly safe.

JaroslavTulach commented on engine/language-server/src/main/resources/application.conf:
```diff
@@ -46,6 +46,7 @@ logging-service {
     org.enso.languageserver.runtime.RuntimeConnector = debug
     org.enso.interpreter.runtime.HostClassLoader = error
     java.lang.ProcessBuilder = warn
+    Standard.Base.Logging.Progress = error
```
- yes, by default suppress progress logging to file and console of progress messages
- of all severity levels below `error`

JaroslavTulach reviewed (commented):
- How does one enable the logging of progress events to console?
- Is there a CLI option to pass to `enso` executable and see those messages?

hubertp committed:
Prioritize env vars for log config

An entry in logger's config always overriden config provided by env
vars, on name conflict. This was against the spec.

hubertp commented:
> How does one enable the logging of progress events to console?
> Is there a CLI option to pass to enso executable and see those messages?

Yes, https://github.com/enso-org/enso/blob/develop/docs/infrastructure/logging.md#custom-log-levels.
Unfortunately there was a bug if an entry already existed in the config file (as it is with this PR). This has been fixed in https://github.com/enso-org/enso/pull/13550/commits/911417a765937df3d6a78cd473b92a5eb49a23ff
It's as simple as providing `-DStandard.Base.Logging.Progress.Logger.level=trace` to get the old behaviour.

<blockquote><img src="https://opengraph.githubassets.com/f84c29cdb071268185e055ca31af2b6b0032c64464a568c1497b29f773d43ba7/enso-org/enso" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/enso-org/enso/blob/develop/docs/infrastructure/logging.md">enso/docs/infrastructure/logging.md at develop · enso-org/enso</a></strong></div><div>Enso Analytics is a self-service data prep and analysis platform designed for data teams. - enso-org/enso</div></blockquote>

hubertp committed:
Fix NI build for logs

Once we switched to NI, logging has lagged behind.
NI initialized logging at build time which meant that no env
configuration was allowed during runtime.
This changes ensures that Enso's logging is initialized at runtime.

hubertp commented:
Thanks @JaroslavTulach for pointing out problems in NI. With https://github.com/enso-org/enso/pull/13550/commits/25a8f89e220ff2c63dd29ef9a637194ba3d9eb64 it now works the same in JVM and NI mode.

JaroslavTulach reviewed (approved):
It'd be good to have an _integration test_ that changes the logging level and verifies changes are applied. A test like 

https://github.com/enso-org/enso/blob/209a7cb2fb8b934f6e5ec7c369139df2de0ad8a5/test/Base_Tests/src/System/Runner_Spec.enso#L60

that is executed both - in `--jvm` as well as _native image_ mode.

hubertp committed:
Merge branch 'develop' into wip/hubert/13368-progress-logging

hubertp committed:
Basic test for custom log level via env var

Enhanced a runner test so that simple log messages are present when log
level is set explicitly.
